addons/stock/models/stock_location.py

```diff
@@ -466,3 +466,13 @@ def toggle_active(self):
         for route in self:
             route.with_context(active_test=False).rule_ids.filtered(lambda ru: ru.active == route.active).toggle_active()
         super().toggle_active()
+
+    @api.constrains('company_id', 'rule_ids')
```

Comment:
very unlikely to happen but if you manage to update the company of a stock rule already set to a route you won't trigger the constrains. I would do 
```suggestion
    @api.constrains('company_id', 'rule_ids.company_id')
```